src/jbuf.c

```diff
@@ -515,62 +622,27 @@ int jbuf_get(struct jbuf *jb, struct rtp_header *hdr, void **mem)
 	mtx_lock(jb->lock);
 	STAT_INC(n_get);
 
-	if (jb->nf <= jb->wish || !jb->packetl.head) {
-		DEBUG_INFO("not enough buffer packets - wait.. "
-			   "(n=%u wish=%u)\n", jb->n, jb->wish);
-		STAT_INC(n_underflow);
-		plot_jbuf_event(jb, 'U');
-		err = ENOENT;
-		goto out;
-	}
-
-	/* When we get one packet P[i], check that the next packet P[i+1]
-	   is present and have a seq no. of seq[i] + 1.
-	   If not, we should consider that packet lost. */
-
 	f = jb->packetl.head->data;
 
-#if JBUF_STAT
-	/* Check sequence of previously played packet */
-	if (jb->seq_get) {
-		const int16_t seq_diff = f->hdr.seq - jb->seq_get;
-		if (seq_less(f->hdr.seq, jb->seq_get)) {
-			DEBUG_WARNING("get: seq=%u too late\n", f->hdr.seq);
-		}
-		else if (seq_diff > 1) {
-			STAT_ADD(n_lost, 1);
-			plot_jbuf_event(jb, 'T');
-			DEBUG_INFO("get: n_lost: diff=%d,seq=%u,seq_get=%u\n",
-				   seq_diff, f->hdr.seq, jb->seq_get);
-		}
-	}
-#endif
+	uint32_t next_playout = (uint32_t)(tmr_jiffies() * jb->p.srate / 1000);
 
-	/* Update sequence number for 'get' */
-	jb->seq_get = f->hdr.seq;
+	/* Check playout time */
+	if (f->playout_time > next_playout) {
```

Comment:
Frame completeness is not checked. Is it already ensured by skew and jitter?

Reply:
Yes, it's ensured by jitter compensation, it's now a time based buffering delay, this should give the lowest delay which is needed to buffer.

Reply:
okay, we can compare the plots if you are ready.

Reply:
Sure, give me some time to prepare and ensure the algorithms are working correctly (wrap-around calculations are tough ;-))